bundle/src/main/java/com/adobe/acs/tools/dumplibs/DumplibsServlet.java

```diff
@@ -0,0 +1,217 @@
+package com.adobe.acs.tools.dumplibs;
+
+import com.adobe.granite.ui.clientlibs.ClientLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibraryManager;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.commons.json.JSONArray;
+import org.apache.sling.commons.json.JSONException;
+import org.apache.sling.commons.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@SlingServlet(
+        label = "ACS AEM Tools - Dumplibs",
+        description = "ACS AEM Tools End-point for Dumplibs",
+        methods = {"GET"},
+        paths = {"/bin/acs-tools/dumplibs"},
```

Comment:
This should be bound to the resource type of `/etc/acs-tools/dumplibs` and use that resource as the entry point. Generally speaking, path-based servlets are an anti-pattern.

If needed you can have this response to a selector to isolate this servlets behavior.

Reply:
@davidjgonzalez sure, I can make that change

Reply:
@davidjgonzalez actually, I don't see the issue with using path based servlet for this purpose. This servlet has nothing to do with the resource itself. The servlet calls the `HtmlLibraryManager` service exclusively and uses nothing of the resource, or any other resources for that matter. As such I think the use of path here is warranted and appropriate.

If your `anti-pattern` argument was referencing this: [https://sling.apache.org/documentation/the-sling-engine/servlets.html#caveats-when-binding-servlets-by-path](https://sling.apache.org/documentation/the-sling-engine/servlets.html#caveats-when-binding-servlets-by-path); none of the caveats apply in this case.

Let me know your thoughts, 
thanks!

Reply:
using the path-based approach introduces a new (virtual) unneeded virtual path that is also (more) difficult to permission.

Basically, there's no good reason TO introduce this path. You already have a resource (/etc/acs-tools/dumplibs) that represents your tool, which is what the HTTP requests (that are the actions for this tool) should interact with... 

The servlet has everything to do w/ the resource since the resource represents your tool (and to be clear your tool != HtmlLibraryManager). Most servlet end-points just call other Services to do the work (or atleast they probably should), but they still are bound to resources.

Hope that helps clarify?

Reply:
@davidjgonzalez, while I disagree on the point regarding this being a "unneeded virtual path" I do agree on the permission point, even thought not extremely relevant in this specific case.

I *dont mind*  registering this servlet with a `resouceType`, and to avoid further discussion and delays I've implemented it, please see my latest commits :)

> I still prefer the path, in this very specific case, as it is more explicit for an external tool and other reasons I mentioned above, but that's just me :)  `resouceType` based servlets are usually the way to go, when it makes sense, which is most cases, but this isn't most cases :)